CMakeLists.txt

```diff
@@ -2,7 +2,8 @@
 # Licensed under the MIT license.
 
 # set minimum cmake version
-cmake_minimum_required(VERSION 3.14 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.15 FATAL_ERROR)
+cmake_policy(SET CMP0092 NEW)
```

Comment:
This policy, which is present in v3.15, removes "/W3" from CMAKE_CXX_FLAGS. It's been a long-standing nuisance that the CMake devs added this flag, because when adding /W4 and /WX to a target, MSVC++ complains the both W3 and W4 are on the command line.

By upgrading, and ensuring the policy is on, W3 will no longer be added to generated makefiles.